ui/job-view/details/BugFiler.jsx

```diff
@@ -788,7 +791,7 @@ export class BugFilerClass extends React.Component {
             </form>
           </ModalBody>
           <ModalFooter>
-            <Button color="secondary" onClick={this.submitFiler}>
+            <Button color="secondary" disabled={!selectedProduct} onClick={this.submitFiler}>
```

Comment:
I've been unable to figure out why clicking submit was saying this.state was undefined, but this effectively works around that problem by just not allowing the Submit button to be clicked unless a product is already selected. I wasn't able to figure out how to conditionally show a tooltip on the disabled button to help direct users to the product search box, so this is just silently disallowing the submit from happening.

Reply:
In order for ``this`` to be accessible in a function, you'd need to do 1 of 3 things.  The one we're leaning toward at this point is class-level-functions (line 359):

    submitFiler = async () => {

The second one would be to ``bind`` the function to ``this`` in the constructor like:

    this.submitFiler = this.submitFiler.bind(this);

But we're moving away from doing this.

And the last would be to use the first syntax in the ``render()`` function like:

    <Button color="secondary" onClick={() => this.submitFiler()}>

But this creates a new function each time render is called which is slightly less efficient.  Not a huge deal, but the first pattern is the desired one.  However, this last one IS the right one in the case where you want to pass the event to the function so that you could use it somehow (like look if it's a ctrl+click or something):

    <Button color="secondary" onClick={ev => this.submitFiler(ev}>

Sorry for being long-winded, but figured it'd be interesting just for your information.  :)